### ISSUE TITLE
Can one run validation after N training iterations instead of N epochs ?

### ISSUE THREAD

**Issue by BetFab** [pytorch/fairseq#2414]
#### Can one run validation after N training iterations instead of N epochs ?
I am training a **transformer** for a **translation** task.  
My train set is somehow big (50 millions samples) and I only have 2 GPUs available for training.   
Currently, I am evaluating my model on a validation set every epoch but one epoch is very long to proceed.   
I was wondering if it was possible with fairseq to checkpoint and evaluate a model every half epoch for instance; aka every N batchs or iterations ?

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.6.0 
 - OS (e.g., Linux): Linux Ubuntu 18.04
 - How you installed fairseq (`pip`, source): source 
 - Build command you used (if compiling from source): pip install --editable ./
 - Python version: 3.6.9
 - CUDA/cuDNN version: 10.1 / 7.4.2

**Comment by BetFab** [comment]
option `--save-interval-updates` for training does exactly that.

**Comment by aflah02** [comment]
Hey how exactly were you specifying how many epochs directly? I've been looking for a suitable command line argument but can't seem to find it. Everything seems to be dealing with something called updates and since I'm new to fairseq i don't quite get how to use them
